Develop

HarijsST opened the pull request.

oskarslinde committed:
Added CalculatorResultModel + added some tests

HarijsST committed:
update CalculatorResultModel
update salaryCalulation

HarijsST committed:
Create Animal Park

HarijsST committed:
Update Animal Park
Canine/Feline extends Animal
Dog/Wolf extends Canine
Cat/Lion extends Feline

HarijsST committed:
Update Animal Park
Rename AnimalApp to KingdomApp

oskarslinde commented on hBank/src/main/java/animalPark/animals/Wolf.java:
```diff
@@ -0,0 +1,11 @@
+import animalPark.AnimalInterface;
+import animalPark.Canine;
+
+public class Wolf extends Canine implements AnimalInterface {
```
For this and other classes - Dog, Cat and Lion.
How we can make sure that other classes can't extends these classes any further?

oskarslinde commented on hBank/src/main/java/animalPark/Feline.java:
```diff
@@ -0,0 +1,7 @@
+package animalPark;
+
+public class Feline extends Animal{
```
How can we make sure that it's not possible to instantiate objects of this class?

oskarslinde replied:
Done - abstract class

oskarslinde commented on hBank/src/main/java/animalPark/AnimalInterface.java:
```diff
@@ -0,0 +1,5 @@
+package animalPark;
+
+public interface AnimalInterface {
```
Interface names should be adjectives. They should end with “able” or “ible” whenever the interface provides a capability; otherwise, they should be nouns. Interface names follow the same capitalization convention as class names: public interface Serializable {...} public interface SystemPanel {...}

For this case I would suggest to use something like "Noisy", or "Soundful"

oskarslinde replied:
Done

oskarslinde commented on hBank/src/main/java/animalPark/KingdomApp.java:
```diff
@@ -0,0 +1,36 @@
+import animalPark.animals.Wolf;
+
+public class KingdomApp {
+    public static void main(String[] args) {
```
Create a instance method that generates 5 (of your choice) objects and returns list of them as List<AnimalInterface>
additionally add a test that validates this method

oskarslinde replied:
Done

HarijsST committed:
Update Animal Park

oskarslinde committed:
Continue work with AnimalPark + add AnimalRepository

HarijsST committed:
Update Animal Park
- all Animals have toString method;
- create getFirstAnimalfromList
- create getLastAnimalFromList

HarijsST committed:
Update Animal Park

oskarslinde committed:
Continue work with AnimalPark

HarijsST committed:
Update Animal Park
Create Student List

HarijsST committed:
Update Student List/StudentBio/toString

HarijsST committed:
Update Animal

oskarslinde commented:
Ir ok
